### PR TITLE
Add deprecation for filter getters with bool reference

### PR DESCRIPTION
**Pull request by tin1254**
Discussed in https://github.com/PointCloudLibrary/pcl/pull/4828#discussion_r663296844

Deprecation is added in `pcl_cuda::filter` and `pcl::VoxelGrid`

### CODE REVIEW

**Review by mvieth — COMMENTED** [review]
The functions should stay until 1.16, not 1.15. It was 1.15 before the 1.12 release, so now (after the 1.12 release) it should be 1.16

File: cuda/filters/include/pcl/cuda/filters/filter.h
Comment (tin1254) [reply cuda/filters/include/pcl/cuda/filters/filter.h:134]: Done.

For the documentation, I copied it from the other `getFilterLimitsNegative` for consistence

**Comment by kunaltyagi** [comment]
Target 1.12.1 or 1.13?

**Review by mvieth — APPROVED** [review]

**Comment by mvieth** [comment]
> Target 1.12.1 or 1.13?

I don't see a reason why 1.12.1 would not be possible, so 1.12.1?

**Comment by tin1254** [comment]
> > Target 1.12.1 or 1.13?
> 
> I don't see a reason why 1.12.1 would not be possible, so 1.12.1?

`PCL_DEPRECATED(1, 12.1, ...)` doesn't compile, maybe 1.13?

**Comment by mvieth** [comment]
Sorry for the misunderstanding: `PCL_DEPRECATED(1, 16, ...)` was correct. We want to give everyone enough time to switch to the other function before removing this function.
The question about 1.12.1 or 1.13 was, whether we merge this pull request immediately (then it would be included in the 1.12.1 release) or we merge it after we released 1.12.1, then it would be included in the 1.13.0 release. One reason why we don't merge some pull requests immediately is because we want to keep 1.12.0 and 1.12.1 ABI and API compatible. Adding a deprecation warning and documentation breaks neither ABI nor API, so I think this can be merged right after everyone approves.
Please undo your most recent changes, sorry again for the misleading formulation.

**Comment by tin1254** [comment]
Thanks for clarifying it, I'm not so familiar with the releasing procedure 😅

**Review by kunaltyagi — APPROVED** [review]

**Pull request merged**
